frontend/apps/web/components/site-header/AccountStatusHandler.tsx

```diff
@@ -35,3 +45,69 @@ export function AccountStatusHandler(props: Props) {
     </div>
   );
 }
+
+interface TrialCountdownProps {
+  createdDate: number;
+  isAccountStatusValidResp: IsAccountStatusValidResponse | undefined;
+}
+
+const TRIAL_DURATION_DAYS = 14;
+
+// TODO: created_at is coming back as blank
+
+function TrialCountdown(props: TrialCountdownProps) {
+  const { createdDate, isAccountStatusValidResp } = props;
+
+  const now = Date.now();
+  const trialEndDate = new Date(
+    createdDate + TRIAL_DURATION_DAYS * 24 * 60 * 60 * 1000
+  );
+  const daysRemaining = Math.max(
+    0,
+    Math.ceil((trialEndDate.getTime() - now) / (1000 * 60 * 60 * 24))
+  );
```

Comment:
Would be cleaner to use date-fns to do most of this for you so that it properly handles time changes like daylight savings, etc.

An example of something more functional:

```
import { differenceInDays, addDays, isAfter } from 'date-fns';


interface TrialStatus {
  isTrialExpired: boolean;
  isNearingEnd: boolean;
  daysRemaining: number;
  trialEndDate: Date;
}

export function checkTrialStatus(createdAt: Date): TrialStatus {
  const TRIAL_DURATION_DAYS = 14;
  const WARNING_THRESHOLD_DAYS = 3;
  
  const currentDate = new Date();
  const trialEndDate = addDays(createdAt, TRIAL_DURATION_DAYS);
  
  const daysRemaining = differenceInDays(trialEndDate, currentDate);
  
  return {
    isTrialExpired: isAfter(currentDate, trialEndDate),
    isNearingEnd: daysRemaining <= WARNING_THRESHOLD_DAYS && daysRemaining > 0,
    daysRemaining: Math.max(0, daysRemaining),
    trialEndDate
  };
}
```

And then to use it:
```ts
const trialStatus = checkTrialStatus(createdAt);

if (trialStatus.isTrialExpired) {
  console.log('Trial has expired');
} else if (trialStatus.isNearingEnd) {
  console.log(`Trial ending soon! ${trialStatus.daysRemaining} days remaining`);
} else {
  console.log(`${trialStatus.daysRemaining} days remaining in trial`);
}
```

Reply:
I don't think this handles the last day (hours remaining) but you kind of get the point here.